src/idl/include/idl/tree.h

```diff
@@ -378,6 +380,7 @@ struct idl_enumerator {
      mapped to a native data type capable of representing a maximally-sized
      enumeration */
   IDL_ANNOTATABLE(uint32_t) value;
+  IDL_ANNOTATABLE(bool) is_default;
```

Comment:
In closing the `enum` declaration we should check if there's only one default. Plus, I'm thinking it may actually make more sense to have a `default_enumerator` member in the `enum` too/instead?

Reply:
The `default_enumerator` could then point to the first enumerator is not explicit default is provided? (Probably works like that, but I'm not a 100% sure)

Reply:
Maybe introduce `IDL_DEFAULT_ENUMERATOR` that equals `IDL_ENUMERATOR | 1u`. The (implicit) default case for unions works like that as well. Don't know if it makes sense, but it might be worth looking into that.

Reply:
My attempt here was to modify the existing interface as little as possible, but your approach should result in less duplicated functionality, as the different backends will not have to implement their own checks

Reply:
I will look into the default member approach

Reply:
Changed to a default_enumerator pointer in the idl_enum class